### PR TITLE
feat(web-components): #734 add support for an icon inside of ic-tooltip

### PR DESCRIPTION
**Pull request by ab2583**
## Summary of the changes
I have added support for an icon inside of tooltips. When using an icon, the label of the tooltip becomes left aligned instead of centered, and the tooltip stays inline with the first line of text.

You are able to provide some alt text for icon via the `icon-alt-text` prop. If you do not provide alt text the icon is hidden from SRs, but if alt text is given, then the icon is available to SRs.

## Related issue
#734 

## Checklist

(I'm currently working my through this)

### General 

- [X] Changes to docs package checked and committed.
- [X] All acceptance criteria reviewed and met. 

### Testing

- [X] Relevant unit tests and visual regression tests added. 
- [X] Visual testing against Figma component specification completed. 

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [ ] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [ ] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [ ] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.

### Screenshots
Screenshot from 2024-05-22 14-01-07

### CODE REVIEW

File: packages/web-components/src/components/ic-tooltip/ic-tooltip.tsx [hunk 1]
```diff
@@ -57,6 +57,11 @@ export class Tooltip {
    */
   @Prop({ reflect: true }) target?: string;
 
+  /**
+   * Alt text to be used with an icon in the icon slot.
```
Comment (gd2910) [inline packages/web-components/src/components/ic-tooltip/ic-tooltip.tsx:61]:
```suggestion
   * The alt text to be used alongside a slotted icon.
```

File: packages/web-components/src/components/ic-tooltip/ic-tooltip.tsx [hunk 2]
```diff
@@ -85,6 +90,11 @@ export class Tooltip {
   }
 
   componentDidRender(): void {
+    if (this.iconAltText && !this.hasIconSlot) {
+      console.warn(
+        "Prop iconAltText is redundant as no icon has been provided"
```
Comment (gd2910) [inline packages/web-components/src/components/ic-tooltip/ic-tooltip.tsx:95]:
```suggestion
        "Prop iconAltText on ic-tooltip is redundant as no icon has been provided to the component"
```

File: packages/web-components/src/components/ic-tooltip/ic-tooltip.tsx [hunk 3]
```diff
@@ -265,16 +275,33 @@ export class Tooltip {
...
           aria-hidden={`${silent}`}
         >
+          {this.hasIconSlot() && (
+            <div
+              class="icon-container"
+              aria-hidden={`${!this.iconAltText}`}
+              aria-label={this.iconAltText}
+              role="img"
+            >
+              <slot name="icon" aria-hidden="true" />
```
Comment (gd2910) [inline packages/web-components/src/components/ic-tooltip/ic-tooltip.tsx:302]: Consider 'tooltip-icon' naming convention to help scope and debug.

File: packages/docs/docs.json
Comment (gd2910) [inline packages/docs/docs.json:1]: Please remove this file from commit.

File: packages/web-components/src/components/ic-tooltip/ic-tooltip.stories.mdx [hunk 4]
```diff
@@ -329,3 +329,33 @@ import readme from "./readme.md";
...
 </Canvas>
+
+### With icon
+
+<Canvas>
+  <Story name="With icon" height="100px">
+    {(args) =>
+      html`<ic-tooltip
+        label="This is a tooltip with an icon and quite a large label to to show the icon aligns with the first line of text."
+        id="ic-tooltip-test-button-default"
```
Comment (gd2910) [inline packages/web-components/src/components/ic-tooltip/ic-tooltip.stories.mdx:340]: Please include iconAltText prop as best use case and screen reader testing.

**Review by gd2910 — COMMENTED** [review]
Thank you very much for your submission, I've left a few comments and I think @GCHQ-Designer-777 will need to have a quick look :)

**Comment by GCHQ-Designer-777** [comment]
> Thank you very much for your submission, I've left a few comments and I think @GCHQ-Designer-777 will need to have a quick look :)

In terms of design I'm happy to approve.

**Comment by GCHQ-Developer-112** [comment]
Hi @ab2583, there's been no activity on this PR for a while. Please leave a comment if you're still working on it

**Comment by GCHQ-Developer-112** [comment]
Hi @ab2583, as this PR has been inactive for 2 months, I'm going to close this PR. Feel free to contribute to our project in the future 😃

**Pull request closed without merging**